Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImage.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+
+#include "TestMain.h"
+#include <WebKitImagePrivate.h>
```

Comment:
Can we test this with images created from WebKit instead of using private api from unit tests?

Reply:
It's just that I don't understand the api, and since it's not used yet by webkit itself, it's even more difficult for me to understand. When I saw a completion handler in the constructor I thought it was a destroy notify to release any data attached to the image, but it's a function to provide the data? Why do we need a function for that? and why do we use completion handler?

Reply:
Is there any wip patch where this is used? Because I still don't understand it.